tests/run-make/issue-107205/main.rs

```diff
@@ -0,0 +1,13 @@
+trait Marker {}
```

Comment:
This test case is different from the original issue reproducer and compiles on current stable unless I'm missing something important: https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=0bb75ef9d55af8608afe6ddfaa054bd3

Reply:
I was trying to reduce repro so that there would be no dependencies on stdlib traits. By the time I started working on this ~2 weeks ago this code was failing on current master d117135f5a9 on aarch64 apple. I can't recall whether I tested it in playground or not though. I've checked that original repro still causing linker to fail, so I'm going to deal with linux issues first and then get back to the repro. Thanks for pointing this out!